grype/matcher/rpmdb/matcher.go

```diff
@@ -84,6 +87,7 @@ func (m *Matcher) matchBySourceIndirection(store vulnerability.ProviderByDistro,
 
 	// use the source package name
 	indirectPackage.Name = sourcePackageName
+	indirectPackage.Version = groupMatches[2]
```

Comment:
now that the regex has useful named capture groups we could leverage https://github.com/anchore/grype/blob/main/internal/parse.go#L6 (I don't think we had this util originally and is an opportune time to swap out for it)